src/data/common/Dhis2DataStoreDataForm.ts

```diff
@@ -217,11 +217,14 @@ export class Dhis2DataStoreDataForm {
         this.subNationals = config.subNationals;
     }
 
-    static async build(api: D2Api): Promise<Dhis2DataStoreDataForm> {
+    static async build(api: D2Api, dataSetCode?: string): Promise<Dhis2DataStoreDataForm> {
         if (cachedStore) return cachedStore;
 
-        const dataStore = api.dataStore(dataStoreNamespace);
-        const storeValue = await dataStore.get<object>(Namespaces.AUTOGENERATED_FORMS).getData();
+        let dataStore = api.dataStore(dataStoreNamespace);
```

Comment:
Normally we try to avoid in place updates. We can use `const` here and then validate if `dataSetCode` does not exist.

```ts
if (!dataSetCode) throw Error(`Unable to load configuration: dataSetCode not defined`);
```

After that passing dataSetCode to load the config:

```ts
const storeValue = await dataStore.get<object>(dataSetCode).getData();
```

What do you think?